inference-engine/src/mkldnn_plugin/nodes/mkldnn_gather_node.cpp

```diff
@@ -44,92 +36,98 @@ MKLDNNGatherNode::MKLDNNGatherNode(const std::shared_ptr<ngraph::Node>& op, cons
                          {{TensorDescCreatorTypes::ncsp, dataPrecision}},
                          impl_desc_type::ref_any);
 }
 
-template <typename index_t, class Conversion>
-void MKLDNNGatherNode::gather() {
-    size_t src_indexSize = getParentEdgeAt(GATHER_INDEXES)->getBlob()->size();
-    size_t outputSize = getChildEdgeAt(0)->getBlob()->byteSize();
-    const auto *src_index = reinterpret_cast<const index_t *>(getParentEdgeAt(GATHER_INDEXES)->getMemoryPtr()->GetPtr());
-    const auto *src_dataDict = reinterpret_cast<const uint8_t *>(getParentEdgeAt(GATHER_DICTIONARY)->getMemoryPtr()->GetPtr());
-    auto *dst_data = reinterpret_cast<uint8_t *>(getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+void MKLDNNGatherNode::createPrimitive() {
+    auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    auto& srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
+        IE_THROW() << errorPrefix_ << " has not allocated destination memory.";
+    if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
+        IE_THROW() << errorPrefix_ << " has not allocated input memory.";
+    if (getSelectedPrimitiveDescriptor() == nullptr)
+        IE_THROW() << errorPrefix_ << " has unidentified preferable primitive descriptor.";
+
+    const SizeVector srcDims = getParentEdgeAt(GATHER_DATA)->getDims().ToSizeVector();
+    const SizeVector idxDims = getParentEdgeAt(GATHER_INDEXES)->getDims().ToSizeVector();
+    const SizeVector dstDims = getChildEdgeAt(0)->getDims().ToSizeVector();
+
+    indexRange = srcDims[axis];
+    batchSize = std::accumulate(srcDims.begin(), srcDims.begin() + batchDims, 1, std::multiplies<size_t>());
+    outerSize = std::accumulate(srcDims.begin() + batchDims, srcDims.begin() + axis, 1, std::multiplies<size_t>());
+    dataLength = std::accumulate(srcDims.begin() + axis + 1, srcDims.end(), 1, std::multiplies<size_t>());
+    srcBatchStride = std::accumulate(srcDims.begin() + batchDims, srcDims.end(), 1, std::multiplies<size_t>());
+    idxBatchStride = std::accumulate(idxDims.begin() + batchDims, idxDims.end(), 1, std::multiplies<size_t>());
+    dstBatchStride = std::accumulate(dstDims.begin() + batchDims, dstDims.end(), 1, std::multiplies<size_t>());
+
+    if (dataLength == 0)
+        IE_THROW() << errorPrefix_ << "had incorrect input parameters dimension!";
+}
 
-    size_t len = dataLength * getParentEdgeAt(GATHER_DICTIONARY)->getDesc().getPrecision().size();
+void MKLDNNGatherNode::execute(mkldnn::stream strm) {
+    const int32_t* srcIndexes = reinterpret_cast<const int32_t*>(getParentEdgeAt(GATHER_INDEXES)->getMemoryPtr()->GetPtr());
+    const uint8_t* srcData = reinterpret_cast<const uint8_t*>(getParentEdgeAt(GATHER_DATA)->getMemoryPtr()->GetPtr());
+    uint8_t* dstData = reinterpret_cast<uint8_t*>(getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+    const size_t dataSize = getSelectedPrimitiveDescriptor()->getConfig().inConfs[GATHER_DATA].desc.getPrecision().size();
```

Comment:
If the variable is constant parameter I would advise to initialize it inside createPrimitive(...). The second question why you are taking precision from selected config but not from the edge? This seems inconsistent comparing with shapes handling.

Reply:
Moved and replaced to the edge. Thank you!